src/celeritas/optical/ImportedModelAdapter.cc

```diff
@@ -49,12 +49,10 @@ ImportedModels::ImportedModels(std::vector<ImportOpticalModel> models)
         CELER_VALIDATE(model.model_class != IMC::size_,
                        << "Invalid imported model class for optical model id '"
                        << model_id << "'");
-        CELER_VALIDATE(
-            std::all_of(model.mfp_table.begin(),
-                        model.mfp_table.end(),
-                        [](auto const& v) { return static_cast<bool>(v); }),
-            << "Import optical model id '" << model_id << "' ("
-            << to_cstring(model.model_class) << ") has invalid MFP vectors");
+
+        // Model MFP vectors may be empty, indicating the model should attempt
+        // to build from optical material data.
```

Comment:
Yes - they can be mixed. I'm not familiar with how actual experiments provide data for Rayleigh scattering, but Geant4 will attempt to use `kRAYLEIGH` mean free paths if provided, and will then attempt to build them from material parameters if they're not present, so this will be consistent with that behavior.

My logic is that all imported model data should have a MFP vector for every optical material, where empty ones indicate that no data has been imported for that material and the model should attempt to build it. Because MFP vectors are present for every material, I think it makes explicit which are imported vs which need to be built. Individual models can then validate if they can import or build MFPs for every optical material.